ubiquity/dell-bootstrap.py

```diff
@@ -760,6 +761,16 @@ def prepare(self, unfiltered=False):
             self.log("rec_type %s, stage %d, device %s" % (rec_type, self.stage, self.device))
             if (rec_type == 'factory' and self.stage == 2) or rec_type == 'hdd':
                 self.fixup_factory_devices(rec_part)
+            if rec_type == 'hdd':
+                # copy old mok key so that user don't need to enroll it again.
+                rootfs = mount[0:-1] + '3'
+                self.log("old rootfs from %s" % rootfs)
+                misc.execute_root('mount', '-o', 'ro', rootfs, '/mnt')
+                if os.path.exists('/mnt/var/lib/shim-signed/mok/MOK.priv') and os.path.exists('/mnt/var/lib/shim-signed/mok/MOK.der'):
+                    misc.execute_root('cp', '-f', '/mnt/var/lib/shim-signed/mok/MOK.der','/tmp')
+                    misc.execute_root('cp', '-f', '/mnt/var/lib/shim-signed/mok/MOK.priv','/tmp')
+                self.log("%s" % subprocess.run('md5sum /tmp/MOK*',shell=True,stdout=subprocess.PIPE))
```

Comment:
`hashlib`?

Reply:
printing md5sum here is convenient for debugging, which engineer can easily get this information from log. I do not tend to use hashlib and it will need more lines of code.

Reply:
Is it really that much more?
```
supermario@W103M466Y2:~$ python3
>>> import hashlib
>>> with open('blah','rb') as f:
...     hashlib.md5(f.read()).hexdigest()
...
'd41d8cd98f00b204e9800998ecf8427e'
>>>

supermario@W103M466Y2:~$ md5sum blah
d41d8cd98f00b204e9800998ecf8427e  blah
```

Reply:
the idea is to keep as little change as possible. but sure can use hashlib to do that. the change is there in commit a24810f